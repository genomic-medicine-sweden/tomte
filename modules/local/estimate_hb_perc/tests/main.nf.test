nextflow_process {

    name "Test Process ESTIMATE_HB_PERC"
    script "../main.nf"
    process "ESTIMATE_HB_PERC"
    config "./nextflow.config"
    tag "modules_local"
    tag "modules"
    tag "estimate_hb_perc"

    setup {
        run("GUNZIP", alias: "GUNZIP_FASTA") {
            script "../../../nf-core/gunzip/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [ id:'fasta' ],
                    file("${projectDir}/test_data/grch37_chr21.fa.gz", checkIfExists: true)
                ])
                """
            }
        }

        run("GUNZIP", alias: "GUNZIP_GTF") {
            script "../../../nf-core/gunzip/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [ id:'gtf' ],
                    file("${projectDir}/test_data/grch37_chr21.gtf.gz", checkIfExists: true)
                ])
                """
            }
        }

        run("STAR_GENOMEGENERATE") {
            script "../../../nf-core/star/genomegenerate/main.nf"
            process {
                """
                input[0] = GUNZIP_FASTA.out.gunzip
                input[1] = GUNZIP_GTF.out.gunzip
                """
            }
        }

        run("STAR_ALIGN") {
            script "../../../nf-core/star/align/main.nf"
            process {
                """
                input[0] = input[0] = Channel.of([[case:'finequagga', id:'ACC5963A1', sample:'ACC5963A1', strandedness:'reverse', sex:'NA', single_end:false, fq_pairs:1, is_fastq:true],
                    [file("${projectDir}/test_data/ACC5963A1_sub_1.fastq.gz", checkIfExists: true),
                    file("${projectDir}/test_data/ACC5963A1_sub_2.fastq.gz", checkIfExists: true)]
                    ])
                input[1] = STAR_GENOMEGENERATE.out.index
                input[2] = GUNZIP_GTF.out.gunzip
                input[3] = false
                input[4] = 'illumina'
                input[5] = false
                """
            }
        }
    }

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = STAR_ALIGN.out.read_per_gene_tab
                input[1] = Channel.of([file("${projectDir}/test_data/subsample.bed", checkIfExists: true)])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
