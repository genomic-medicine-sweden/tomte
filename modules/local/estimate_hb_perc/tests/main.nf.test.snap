{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "case": "finequagga",
                            "id": "ACC5963A1",
                            "sample": "ACC5963A1",
                            "strandedness": "reverse",
                            "sex": "NA",
                            "single_end": false,
                            "fq_pairs": 1,
                            "is_fastq": true
                        },
                        "ACC5963A1_perc_mapping.json:md5,7ddc835f55ee6f4daaec94456b7ad812"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b877b52a91b9f0a47f954b5f27c8d5b1"
                ],
                "json": [
                    [
                        {
                            "case": "finequagga",
                            "id": "ACC5963A1",
                            "sample": "ACC5963A1",
                            "strandedness": "reverse",
                            "sex": "NA",
                            "single_end": false,
                            "fq_pairs": 1,
                            "is_fastq": true
                        },
                        "ACC5963A1_perc_mapping.json:md5,7ddc835f55ee6f4daaec94456b7ad812"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b877b52a91b9f0a47f954b5f27c8d5b1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-15T14:54:34.289923"
    }
}